chainercv/chainer_experimental/datasets/sliceable/sliceable_dataset.py

```diff
@@ -0,0 +1,137 @@
+import six
+
+import chainer
+
+
+def _as_tuple(t):
+    if isinstance(t, tuple):
+        return t
+    else:
+        return t,
+
+
+class SliceableDataset(chainer.dataset.DatasetMixin):
+    """An abstract dataset class that supports slicing.
+
+    This ia a dataset class that supports slicing.
+    A dataset class inheriting this class should implement
+    three methods: :meth:`__len__`, :meth:`keys`, and
+    :meth:`get_example_by_keys`.
```

Comment:
I mean the relationship between `SlicableDataset` and `GetterDataset` should be clear.

From users perspective,
they would first come and read `SlicableDataset`. Since this is not intended to be directly touched by users, we should guide them to `GetterDataset`.

Reply:
Nice suggestion. I agree with you.